server/common/rest.py

```diff
@@ -179,14 +180,21 @@ def diffexp_obs_post(request, data_adaptor):
 
 
 def layout_obs_get(request, data_adaptor):
+    fields = request.args.getlist("layout-name", None)
```

Comment:
Would wrapping this in a `set(...)` help protect against clients/abusers inflating the number of requests?

Reply:
how so?  URLs have a fixed length limit (number of characters), so in practice it can't ever be that big....